client/blocks/login/index.jsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormsButton from 'components/forms/form-button';
+import Card from 'components/card';
+import FormTextInput from 'components/forms/form-text-input';
+import { loginUser } from 'state/login/actions';
+import Notice from 'components/notice';
+import { externalRedirect } from 'lib/route/path';
+
+import {
+	isRequestingLogin,
+	isLoginSuccessful,
+	getError
+} from 'state/login/selectors';
+
+export class Login extends Component {
+	static propTypes = {
+		isRequestingLogin: PropTypes.bool.isRequired,
+		loginUser: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+		isLoginSuccessful: PropTypes.bool,
+		loginError: PropTypes.string,
+		redirectLocation: PropTypes.string,
+		title: PropTypes.string,
+	};
+
+	static defaultProps = {
+		title: '',
+	};
+
+	constructor() {
+		super();
+		this.state = {
+			usernameOrEmail: '',
+			password: '',
+		};
+		this.onChangeField = this.onChangeField.bind( this );
+		this.onSubmitForm = this.onSubmitForm.bind( this );
+	}
+
+	componentDidUpdate() {
+		if ( this.props.isLoginSuccessful ) {
+			externalRedirect( this.props.redirectLocation || '/' );
+		}
+	}
+
+	onChangeField( event ) {
+		this.setState( {
+			[ event.target.name ]: event.target.value
+		} );
+	}
+
+	onSubmitForm( event ) {
+		event.preventDefault();
+		this.props.loginUser( this.state.usernameOrEmail, this.state.password );
+	}
+
+	renderNotices() {
+		if ( this.props.loginError ) {
+			return (
+				<Notice status="is-error" text={ this.props.loginError } />
+			);
+		}
+	}
+
+	render() {
+		const isDisabled = {};
+		if ( this.props.isRequestingLogin ) {
+			isDisabled.disabled = true;
+		}
+
+		return (
+			<div className="login">
+
+				{ this.renderNotices() }
+
+				<form onSubmit={ this.onSubmitForm }>
```

Comment:
Just a thought. Should we add an action in case we have a server side rendering, so this submits to `wp-login.php`? Maybe worry about that when it happens...